core/server/api_container/server/startosis_engine/kurtosis_types/service_config/image_registry_spec.go

```diff
@@ -0,0 +1,167 @@
+	"go.starlark.net/starlark"
+)
+
+const (
+	ImageRegistrySpecTypeName = "ImageRegistrySpec"
+
+	RegistryImageAttr    = "image"
+	RegistryAddrAttr     = "registry"
+	RegistryUsernameAttr = "username"
+	RegistryPasswordAttr = "password"
+)
+
+func NewImageRegistrySpec() *kurtosis_type_constructor.KurtosisTypeConstructor {
+	return &kurtosis_type_constructor.KurtosisTypeConstructor{
+		KurtosisBaseBuiltin: &kurtosis_starlark_framework.KurtosisBaseBuiltin{
+			Name: ImageRegistrySpecTypeName,
+			Arguments: []*builtin_argument.BuiltinArgument{
+				{
+					Name:              RegistryImageAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator: func(value starlark.Value) *startosis_errors.InterpretationError {
+						return builtin_argument.NonEmptyString(value, RegistryImageAttr)
+					},
+				},
+				{
+					Name:              RegistryAddrAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator: func(value starlark.Value) *startosis_errors.InterpretationError {
+						return builtin_argument.NonEmptyString(value, RegistryAddrAttr)
+					},
+				},
+				{
+					Name:              RegistryUsernameAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator: func(value starlark.Value) *startosis_errors.InterpretationError {
+						return builtin_argument.NonEmptyString(value, RegistryUsernameAttr)
+					},
+				},
+				{
+					Name:              RegistryPasswordAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator: func(value starlark.Value) *startosis_errors.InterpretationError {
+						return builtin_argument.NonEmptyString(value, RegistryUsernameAttr)
+					},
+				},
+			},
+		},
+		Instantiate: instantiateImageRegistrySpec,
+	}
+}
+
+func instantiateImageRegistrySpec(arguments *builtin_argument.ArgumentValuesSet) (builtin_argument.KurtosisValueType, *startosis_errors.InterpretationError) {
+	kurtosisValueType, err := kurtosis_type_constructor.CreateKurtosisStarlarkTypeDefault(ImageRegistrySpecTypeName, arguments)
+	if err != nil {
+		return nil, err
+	}
+	return &ImageRegistrySpec{
+		KurtosisValueTypeDefault: kurtosisValueType,
+	}, nil
+}
+
+// ImageRegistrySpec is a starlark.Value that holds all the information to log in to a Docker registry
+type ImageRegistrySpec struct {
+	*kurtosis_type_constructor.KurtosisValueTypeDefault
+}
+
+func (irs *ImageRegistrySpec) Copy() (builtin_argument.KurtosisValueType, error) {
+	copiedValueType, err := irs.KurtosisValueTypeDefault.Copy()
+	if err != nil {
+		return nil, err
+	}
+	return &ImageRegistrySpec{
+		KurtosisValueTypeDefault: copiedValueType,
+	}, nil
+}
+
+// GetImage returns the image that needs to be pulled
+func (irs *ImageRegistrySpec) GetImage() (string, *startosis_errors.InterpretationError) {
```

Comment:
maybe suggest `GetImageName` to be consistent with `ImageBuildSpec` but up to you

Reply:
irs - internal revenue service....